Libraries/Opc.Ua.Client/Session/Session.cs

```diff
@@ -5634,6 +5634,9 @@ ITransportChannel transportChannel
         {
             Utils.LogInfo("Session RECONNECT {0} starting.", SessionId);
 
+            //update endpoint before reconnect
+            m_endpoint.UpdateFromServer();
```

Comment:
this *should* be handled by the session reconnect handler. search for 'm_updateFromServer' and why the condition is not met.

Reply:
UpdateFromServer is quite an expensive operation (open discovery channel, GetEndpoints, etc.) and should only be done if some is not quite right.